cmd/dep/init.go

```diff
@@ -258,6 +262,24 @@ func getProjectData(ctx *dep.Ctx, pkgT gps.PackageTree, cpr string, sm *gps.Sour
 				return projectData{}, errors.Wrap(err, "sm.DeduceProjectRoot") // TODO: Skip and report ?
 			}
 
+			start := time.Now()
+			message := make(chan string, len(v.P.Imports))
+			go func() {
+				goStart := time.Now()
+				defer fmt.Fprintf(os.Stderr, "Go Duration: %s\n", time.Since(goStart))
+				err2 := sm.SyncSourceFor(gps.ProjectIdentifier{ProjectRoot: pr})
+				fmt.Fprintf(os.Stderr, "SyncSourceFor Duration: %s\n", time.Since(goStart))
+				if err2 != nil {
+					message <- fmt.Sprintf("Unable to clone: %s\n", pr)
+				}
+				message <- fmt.Sprintf("Cloning: %s", pr)
+			}()
+
+			select {
+			case m := <-message:
```

Comment:
Ok doing so has increased speed drastically. Though the way Ive written it not sure it needs to be a closure. Im still analyzing the code to make sure the new goroutine to call the closure always has a chance to complete.